src/System.Diagnostics.EventLog/tests/EventLogTraceListenerTests.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogTraceListenerTests
+    {
+        public static IEnumerable<object[]> EventLogData()
+        {
+            yield return new object[] { null, string.Empty };
+            yield return new object[] { new EventLog("logName", ".", "EventLogSource"), "EventLogSource" };
+        }
+
+        [Theory]
+        [MemberData(nameof(EventLogData))]
+        public void EventLogConstructor(EventLog eventLog, string expected)
+        {
+            using (var listener = new EventLogTraceListener(eventLog))
+            {
+                Assert.Equal(expected, listener.Name);
+            }
+        }
+
+        [Fact]
+        public void StringConstructor()
+        {
+            string source = "CustomSource";
+            using (var listener = new EventLogTraceListener(source))
```

Comment:
if it creates the ```EventSource``` on the system. then we might we want to Delete that source after the execution as part of cleanup. https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs#L151

Reply:
It doesn't create any source in the system when creating the listener, you have to create the EventLog/Source manually.